h5netcdf/core.py

```diff
@@ -121,6 +120,72 @@ def __getitem__(self, key):
         return self._h5ds[key]
 
     def __setitem__(self, key, value):
+        # This conditional is used to automatically resize the underlying HDF5
```

Comment:
Rather than automatically resizing if the `value` is too large, I would prefer that the core h5netcdf API match h5py:
- Add a `resize` method that calls the h5py method of the same name.
- Only put automatic resizing in the `netCDF4` compatibility layer.

We could potential expose the `needs_resize()` helper function as part of the public API if you're interested in using it with the non-legacy PAI.